src/experimental/patterns/ec2-app.ts

```diff
@@ -253,6 +253,90 @@ export interface GuEc2AppExperimentalProps extends Omit<GuEc2AppProps, "updatePo
   buildIdentifier: string;
 }
 
+export function rollingUpdatePolicy(maximumInstances: number, minimumInstances?: number): UpdatePolicy {
+  return UpdatePolicy.rollingUpdate({
+    maxBatchSize: maximumInstances,
+    minInstancesInService: minimumInstances,
+    minSuccessPercentage: 100,
+    waitOnResourceSignals: true,
+    /*
+    If a scale-in event fires during an `AutoScalingRollingUpdate` operation, the update could fail and rollback.
+    For this reason, we suspend the `AlarmNotification` process, else availability of a service cannot be guaranteed.
+    Consequently, services cannot scale-out during deployments.
+    If AWS ever supports suspending scale-out and scale-in independently, we should allow scale-out.
+     */
+    suspendProcesses: [ScalingProcess.ALARM_NOTIFICATION],
+    // Note: there is also an important property called pauseTime, but this is set via an Aspect.
+  });
+}
+
+export function modifyRoleForRollingUpdate(scope: GuStack, autoScalingGroup: GuAutoScalingGroup) {
```

Comment:
Could we convert this to a class so that it's picked up by the [metadata aspect](https://github.com/guardian/cdk/blob/main/src/aspects/metadata.ts), allowing us to track usage in case we need to make breaking changes in future?

Similar to `modifyUserDataForRollingUpdate` below too.